### PR TITLE
Add in-context event source creation menu in topology

### PR DESCRIPTION
**Pull request by karthikjeeyar**
Fixes: https://issues.redhat.com/browse/ODC-3057

**Analysis / Root Cause:**
Add a new action menu item that allows for adding Event Soure resources from topology in-context  menu which appears on dragging the edge from the knative service.

**Solution:**
Added the action menu and conditionally rendering the new action menu only if event source CRD are installed and the source node should be a knative service

 **Screenshots:**

incontext-sinkbinding

**_note_**: event source menu item will not appear until you install openshift eventing operator.
 
**unit test coverage report**:
image

image

Browser conformance:

 - [x] Chrome
 - [x] Firefox
 - [ ] Safari
 - [ ]  Edge

cc: @serenamarie125 @christianvogt

### CODE REVIEW

**Comment by invincibleJai** [comment]
@karthikjeeyar i think `sink service` dropdown would be disabled in this scenario. Let's confirm it with @serenamarie125

**Comment by serenamarie125** [comment]
To be consistent with what we are doing with Applications, I'd suggest that we hide the Sink section in this case.
FYI @karthikjeeyar @invincibleJai @christianvogt ^^

**Comment by christianvogt** [comment]
@serenamarie125 I don't agree with hiding this information otherwise this page display zero context as to what service it is sinking to.
When we add in context of an application, the application dropdown is disabled but shows the application name in context. What @karthikjeeyar has done with disabling the knative service dropdown seems reasonable. Although the help text saying to `select a service to sink to` should not be present. Perhaps removed or changed to describe this was the chosen context.

**Comment by serenamarie125** [comment]
ha!  Fair point @christianvogt !  Let's keep it there then - I can look at updating text or something in this situation.  Do you think we can merge the PR as is, and I can handle the messaging in the Final design, and have a follow up PR to address that?  This way we can have more time to test the functionality.

**Comment by christianvogt** [comment]
Yes. Let's go with no help text for now when we disable the service dropdown and do further changes in future pr.

File: frontend/packages/knative-plugin/src/components/add/EventSourcePage.tsx [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
 } from '@console/dev-console/src/components/NamespacedPage';
+import { QUERY_PROPERTIES } from '@console/dev-console/src/const';
 import EventSource from './EventSource';
 
-interface EventSourcePageProps {
-  match: RMatch<{
-    ns?: string;
-  }>;
-}
+export type EventSourcePageProps = RouteComponentProps<{ ns?: string }>;
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/components/add/EventSourcePage.tsx:11]: nit: we don't need to export type

File: frontend/packages/dev-console/src/components/topology/Topology.tsx [hunk 2]
```diff
@@ -51,6 +52,7 @@ interface StateProps {
   filters: TopologyFilters;
   application: string;
   serviceBinding: boolean;
+  eventSourceEnabled: boolean;
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/Topology.tsx:55]: @jeff-phillips-18 do we not have a better way to track feature enablement in topology?

Comment (jeff-phillips-18) [reply frontend/packages/dev-console/src/components/topology/Topology.tsx:55]: Enablement is in State so barring passing it thru all the layers to get from a component into a util there is none. Adding it to the graph data is probably not the best way to store it (things won't get updated on state changes), but that goes for the other stored items that are there now :/. Likely we should remove them all and pass the args around. I'd be OK with a follow-up issue to resolve this.

Comment (karthikjeeyar) [reply frontend/packages/dev-console/src/components/topology/Topology.tsx:55]: Raised a bug to track https://issues.redhat.com/browse/ODC-3506

**Pull request merged**